src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -324,12 +340,76 @@ private void PossiblyGlobArg(string arg, bool usedQuotes)
             }
 #endif // UNIX
 
+            // Check if the start of the path is a filesystem drive
+            if (stringConstantType != StringConstantType.SingleQuoted)
+            {
+                arg = ResolvePath(arg, Context);
+            }
+
             if (!argExpanded)
             {
                 _arguments.Append(arg);
             }
         }
 
+        /// <summary>
+        /// Check if string is prefixed by psdrive, if so, expand it if filesystem path.
+        /// </summary>
+        /// <param name="path">The potential PSPath to resolve.</param>
+        /// <param name="context">The current ExecutionContext.</param>
+        /// <returns>Resolved PSPath if applicable otherwise the original path</returns>
+        internal static string ResolvePath(string path, ExecutionContext context)
```

Comment:
Do we really need the new method? We have so many methods in Globbing that I'd expect we already have appropriate method there. Also there is a support for "~" and provider prefix (why not support `FileSystem::`).

Reply:
I looked and didn't see anything existing.  As for PSProviderPath format, I don't see a need to support that unless users show a use case for it.

Reply:
We could use the Globbing method:
```c#
        internal string GetProviderPath(
            string path,
            CmdletProviderContext context,
            out ProviderInfo provider,
            out PSDriveInfo drive)
```

Reply:
@SteveL-MSFT can `GetProviderPath` be used for this?

Reply:
This seems to be working, will change it and test.